commit dbf568755e7781a4ce8bd3c587c2c388600ae661
Author: Giovanni Cabiddu <giovanni.cabiddu@intel.com>
Date:   Mon Oct 12 21:38:33 2020 +0100

    crypto: qat - register crypto instances based on capability
    
    Introduce the function adf_hw_dev_has_crypto() that returns true if a
    device supports symmetric crypto, asymmetric crypto and authentication
    services.
    If a device has crypto capabilities, add crypto instances to the
    configuration.
    This is done since the function that allows to retrieve crypto
    instances, qat_crypto_get_instance_node(), return instances that support
    all crypto services.
    
    Signed-off-by: Giovanni Cabiddu <giovanni.cabiddu@intel.com>
    Reviewed-by: Wojciech Ziemba <wojciech.ziemba@intel.com>
    Reviewed-by: Fiona Trahe <fiona.trahe@intel.com>
    Reviewed-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/drivers/crypto/qat/qat_common/qat_crypto.c b/drivers/crypto/qat/qat_common/qat_crypto.c
index ab621b7dbd20..089d5d7b738e 100644
--- a/drivers/crypto/qat/qat_common/qat_crypto.c
+++ b/drivers/crypto/qat/qat_common/qat_crypto.c
@@ -117,10 +117,15 @@ int qat_crypto_dev_config(struct adf_accel_dev *accel_dev)
 {
 	int cpus = num_online_cpus();
 	int banks = GET_MAX_BANKS(accel_dev);
-	int instances = min(cpus, banks);
 	char key[ADF_CFG_MAX_KEY_LEN_IN_BYTES];
 	int i;
 	unsigned long val;
+	int instances;
+
+	if (adf_hw_dev_has_crypto(accel_dev))
+		instances = min(cpus, banks);
+	else
+		instances = 0;
 
 	if (adf_cfg_section_add(accel_dev, ADF_KERNEL_SEC))
 		goto err;